cmd/gitops/ui/run/cmd.go

```diff
@@ -110,7 +121,30 @@ func runCmd(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("could not create kube http client: %w", err)
 	}
 
-	profilesConfig := server.NewProfilesConfig(rawClient, options.HelmRepoNamespace, options.HelmRepoName)
+	profileCache, err := cache.NewCache(options.ProfileCacheLocation)
+	if err != nil {
+		return fmt.Errorf("failed to create cacher: %w", err)
+	}
+
+	profileWatcher, err := watcher.NewWatcher(watcher.Options{
+		KubeClient:         rawClient,
+		Cache:              profileCache,
+		MetricsBindAddress: options.WatcherMetricsBindAddress,
+		HealthzBindAddress: options.WatcherHealthzBindAddress,
+		WatcherPort:        options.WatcherPort,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to start the watcher: %w", err)
+	}
+
+	go func() {
+		if err := profileWatcher.StartWatcher(); err != nil {
+			log.Error(err, "failed to start profile watcher")
```

Comment:
this might not be from starting but just the controller running in general right?